src/wallet/api/wallet.cpp

```diff
@@ -197,6 +197,44 @@ bool Wallet::addressValid(const std::string &str, bool testnet)
   return get_account_integrated_address_from_str(address, has_payment_id, pid, testnet, str);
 }
 
+bool Wallet::keyValid(const std::string &key_string, const std::string &address_string, bool isViewKey, bool testnet, std::string &error)
```

Comment:
I'd rename key_string to secret_key_string, since there's no typing to make it error proof here.